homeassistant/components/zeroconf/__init__.py

```diff
@@ -552,12 +552,20 @@ def _first_non_link_local_address(
     """Return the first ipv6 or non-link local ipv4 address, preferring IPv4."""
     for address in addresses:
         ip_addr = ip_address(address)
-        if not ip_addr.is_link_local and ip_addr.version == 4:
+        if (
+            not ip_addr.is_link_local
```

Comment:
Why are we ignoring link-local addresses?  That's a valid IP address in the context of discovering stuff in the LAN.

Reply:
This code only orders the ip addresses to present the one we can most likely connect to first.  All of them are still accessible as `addresses`

Reply:
OK.  First non link local address.  Understood.